src/capi/capi.cc

```diff
@@ -0,0 +1,260 @@
+#include "capi.h"
+#include "statefunctions/statecontroller.h"
+#include "types/settings.h"
+#include "types/gamestate.h"
+#include "controllers/controllermanager.h"
+#include "controllers/playercontroller.h"
+#include <cstring>
+#include <string>
+#include <map>
+#include <vector>
+
+extern "C" {
+
+/**
+* Opaque Type Wrappers
+*/
+
+struct GameStateWrapper {
+    std::unique_ptr<mahjong::GameState> state;
+};
+
+struct PlayerControllerWrapper {
+    std::unique_ptr<mahjong::PlayerController> controller;
+};
+
+/**
+* External Player Controller
+*/
+
+class ExternalPlayerController : public mahjong::PlayerController {
```

Comment:
I feel like we should divide this out into separate implementation files. Probably best to separate the  header too and if we want to make it easy to just include the api.h then we could have the api.h just include all the required api headers directly.